constructors.js

```diff
@@ -108,3 +152,28 @@
    * @param  {Spellcaster} target         The spell target to be inflicted.
    * @return {boolean}                    Whether the spell was successfully cast.
    */
+
+   this.invoke = function(spell, target) {
+    if(spell instanceof Spell) {
+      if(spell instanceof DamageSpell) {
+        if(target instanceof Spellcaster) {
+          if(this.spendMana(spell.cost) === true) {
+            target.inflictDamage(spell.damage);
+            return true;
+          } else {
+            return false;
+          }
+        } else {
+          return false;
+        }
+      } else if (this.spendMana(spell.cost) === true){
```

Comment:
Calling this.spendMana twice here means that regular spells cost twice as much mana.